File: csharpcore/StarterBot/Entities/CellStateContainer.cs
using System.Collections.Generic;
using StarterBot.Enums;

namespace StarterBot.Entities
{
    public class CellStateContainer
    {
        public int X { get; set; } 
        public int Y { get; set; } 
        public PlayerType CellOwner { get; set; } 
        public List<Building> Buildings { get; set; } 
        public List<Missile> Missiles { get; set; } 
    }
}

File: java/src/main/java/za/co/entelect/challenge/entities/CellStateContainer.java
package za.co.entelect.challenge.entities;

import za.co.entelect.challenge.enums.PlayerType;

import java.util.ArrayList;
import java.util.List;

public class CellStateContainer {
    public int x;
    public int y;
    public PlayerType cellOwner;
    protected List<Building> buildings;
    protected List<Missile> missiles;

    public CellStateContainer(int x, int y, PlayerType cellOwner) {
        this.x = x;
        this.y = y;
        this.cellOwner = cellOwner;
        this.buildings = new ArrayList<>();
        this.missiles = new ArrayList<>();
    }

    public List<Building> getBuildings() {
        return this.buildings;
    }

    public List<Missile> getMissiles() {
        return this.missiles;
    }

}
